proxy/src/com/aerospike/client/proxy/grpc/GrpcChannelExecutor.java

```diff
@@ -374,6 +369,58 @@ public void run() {
 	 * Process a single iteration.
 	 */
 	private void iterate() {
+		switch (channelState.get()) {
+			case READY:
+				executeCalls();
+				break;
+
+			case SHUTTING_DOWN:
+				boolean allCallsCompleted = pendingCalls.isEmpty() &&
+					streams.values().stream()
+						.allMatch(grpcStream -> grpcStream.getOngoingRequests() == 0);
+
+				int closeTimeout = grpcClientPolicy.closeTimeout;
```

Comment:
if you set "closetimeout" to Long.MAX_VALUE, when closeTimeout is 0, you can avoid this block by treating timeout as positive number.